src/spikeinterface/core/tests/test_sortinganalyzer.py

```diff
@@ -147,6 +142,9 @@ def _check_sorting_analyzers(sorting_analyzer, original_sorting):
         # unit 1, 3, ... should be removed
         assert np.all(~np.isin(data["result_two"], [1, 3]))
 
+    # test compute with extension-specific params
+    sorting_analyzer.compute(input=["dummy"], dummy=dict(param1=5.5))
```

Comment:
I do not like this both list + dict writting.
The list was done for a shortcut.
Lets discuss this behavior I think this is too confusing.

For me it was:
 * dict
 * str
 * list
 
Here you are introducing list + dict

Reply:
I would also prefer and use this. Or even better, having `extension_params` as a sepratate dictionary, that could be also propagated to the `compute_several_extensions(ext_list, ext_params_dict)`

Reply:
Honestly I like that flow better. list, dict makes more sense rather than nested dict. So if we remove the option to do it with a dict and force the user to either do: 

str: dict
list: dicts

then I'm okay with that flow.